pandas/io/orc.py

```diff
@@ -47,6 +47,8 @@ def read_orc(
         ``file://localhost/path/to/table.orc``.
     columns : list, default None
         If not None, only these columns will be read from the file.
+        Output always follows the ordering of the file and not the columns list.
+        This mirrors the original behaviour of :func:`pyarrow.orc.OrcFile.read`.
```

Comment:
Can you show how this renders?

Reply:
I was thinking about adding ``~`` to the link, could you try that?

Reply:
I think https://github.com/pandas-dev/pandas/pull/49709/files#r1024350615 should allow the `read` sphinx link should allow it to render & link to the pyarrow docs

Reply:
It doesn't create any link (at least in my local environment).

I found this line from the output of `python make.py html`:

    /home/pandas/pandas/io/orc.py:docstring of pandas.io.orc.read_orc:16: WARNING: external py:func reference target not found: pyarrow.orc.OrcFile.read

Maybe I'm missing some configuration?

From a quick search, I found only 2 usages of `external+pyarrow` in the entire project: in [this .rst file](https://github.com/pandas-dev/pandas/blob/main/doc/source/reference/arrays.rst#id77). Their correctly open the respective doc in pyarrow.

Reply:
Maybe it's worth opening another issue for adding links to all references to pyarrow (only in the docstring).

`pandas.io.orc.to_orc` is also missing `external+pyarrow`.
https://github.com/pandas-dev/pandas/blob/2455f88ba955ba16f3d50521b2e7cf3df5dc1f32/pandas/io/orc.py#L107

Reply:
Sounds good, could you open an issue?

Reply:
I double-checked and they all seem to correctly link to external documentation. 

E.g. [param engine_kwargs for `DataFrame.to_orc`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_orc.html): it has a link to pyarrow, but [in the source-code](https://github.com/pandas-dev/pandas/blob/2455f88ba955ba16f3d50521b2e7cf3df5dc1f32/pandas/io/orc.py#L107) it doesn't have `external+pyarrow` in front.

So we should be good.